File: Problems/No.228_Summary_Ranges/summary_ranges.py
class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """

        # parameters
        ranges = []
        temp = ""
        i, Range = 0, None

        # start computing
        while (i < len(nums)):
            
            temp = str(nums[i])
            Range = 1
            while (Range < len(nums)-i):
                if (nums[i+Range] != nums[i]+Range):
                    break
                Range += 1

            if (Range == 1):
                ranges.append(temp)
            else:
                temp += ("->"+str(nums[i+Range-1]))
                ranges.append(temp)
                i += (Range-1)
            i += 1

        return ranges

File: Problems/No.228_Summary_Ranges/summary_ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        
        // parameters
        std::vector<std::string> ranges;
        std::string temp;
        int range;

        // start computing
        for (int i=0; i<nums.size(); i++) {

            temp = std::to_string(nums[i]);
            for (range=1; range<nums.size()-i; range++)
                if (nums[i+range] != nums[i]+range) break;

            if (range == 1) ranges.push_back(temp);
            else {

                temp += ("->"+std::to_string(nums[i+range-1]));
                ranges.push_back(temp);
                i += (range-1);

            }

        }

        return ranges;

    }
};
